database/src/main/java/com/firebase/ui/database/FirebaseIndexArray.java

```diff
@@ -38,6 +38,9 @@
     private FirebaseArray<String> mKeySnapshots;
     private List<DataSnapshot> mDataSnapshots = new ArrayList<>();
 
+    private List<String> mQueuedAdditions = new ArrayList<>();
```

Comment:
@samtstern Dang, those are much better names. Thanks! 😄

I changed the documentation a bit and I think you're confused about `mHasPendingModifyOrDelete`. It only support moves and deletions. This is the big issue with this PR, it doesn't work for changes. See my comment here: https://github.com/firebase/FirebaseUI-Android/pull/633#discussion_r105979080. We would need some sort of gross timer or something to wait for a group of writes. 😢